apps/vaporgui/MainForm.cpp

```diff
@@ -1186,10 +1186,10 @@ void MainForm::createBookmark()
     int  customViewportWidth = vpp->GetValueLong(vpp->CustomFramebufferWidthTag, 0);
     int  customViewportHeight = vpp->GetValueLong(vpp->CustomFramebufferHeightTag, 0);
 
-    const int     iconSize = BookmarkParams::DefaultIconSize();
-    const int     iconDataSize = iconSize * iconSize * 3;
-    unsigned char iconData[iconDataSize];
-    char          iconDataString[64];
+    const int      iconSize = BookmarkParams::DefaultIconSize();
+    const int      iconDataSize = iconSize * iconSize * 3;
+    unsigned char *iconData = new unsigned char[iconDataSize];
```

Comment:
If this pointer is already pointing to a memory block, then this line will cause the old block to leak.

Reply:
suggest avoid managing memory manually by using a unique ptr:
`auto iconData = std::make_unique<unsigned char[]>( iconDataSize );`

Reply:
How could this pointer already be pointing to a memory block?

Reply:
> How could this pointer already be pointing to a memory block?

That's right, it cannot. The original code won't produce a memory leak.